sdk-java/src/test/java/ly/count/sdk/java/internal/sc_MV_ManuelViewTests.java

```diff
@@ -0,0 +1,237 @@
+     * stopViewWithID(x2), addSegmentationToViewWithID, addSegmentationToViewWithName
+     * <p>
+     * called with empty string values
+     * <p>
+     * versions with and without segmentation
+     * <p>
+     * nothing should crash, no events should be recorded
+     */
+    @Test
+    public void MV_101_badValues_emptyString() {
+        Countly.instance().init(TestUtils.getConfigViews());
+        TestUtils.validateEQSize(0);
+        Assert.assertEquals(0, TestUtils.getCurrentRQ().length);
+
+        Countly.instance().view("");
+        Countly.instance().view("", false);
+        Countly.instance().views().startAutoStoppedView("");
+        Countly.instance().views().startAutoStoppedView("", TestUtils.map());
+        Countly.instance().views().startView("");
+        Countly.instance().views().startView("", TestUtils.map());
+        Countly.instance().views().pauseViewWithID("");
+        Countly.instance().views().resumeViewWithID("");
+        Countly.instance().views().stopViewWithName("");
+        Countly.instance().views().stopViewWithName("", TestUtils.map());
+        Countly.instance().views().stopViewWithID("");
+        Countly.instance().views().stopViewWithID("", TestUtils.map());
+        Countly.instance().views().addSegmentationToViewWithID("", TestUtils.map());
+        Countly.instance().views().addSegmentationToViewWithName("", TestUtils.map());
+
+        TestUtils.validateEQSize(0);
+        Assert.assertEquals(0, TestUtils.getCurrentRQ().length);
+    }
+
+    /**
+     * pauseViewWithID, resumeViewWithID, stopViewWithName(x2),
+     * stopViewWithID(x2), addSegmentationToViewWithID, addSegmentationToViewWithName
+     * <p>
+     * called with empty string values
+     * <p>
+     * versions with and without segmentation
+     * <p>
+     * nothing should crash, no events should be recorded
+     */
+    @Test
+    public void MV_102_badValues_nonExistingViews() {
+        Countly.instance().init(TestUtils.getConfigViews());
+        TestUtils.validateEQSize(0);
+        Assert.assertEquals(0, TestUtils.getCurrentRQ().length);
+
+        Countly.instance().views().pauseViewWithID("idv1");
+        Countly.instance().views().resumeViewWithID(TestUtils.keysValues[1]);
+        Countly.instance().views().stopViewWithName(TestUtils.keysValues[2]);
+        Countly.instance().views().stopViewWithName(TestUtils.keysValues[3], TestUtils.map());
+        Countly.instance().views().stopViewWithID(TestUtils.keysValues[4]);
+        Countly.instance().views().stopViewWithID(TestUtils.keysValues[5], TestUtils.map());
+        Countly.instance().views().addSegmentationToViewWithID("idv1", TestUtils.map());
+        Countly.instance().views().addSegmentationToViewWithName(TestUtils.keysValues[1], TestUtils.map());
+
+        TestUtils.validateEQSize(0);
+        Assert.assertEquals(0, TestUtils.getCurrentRQ().length);
+    }
+
+    //(2XX) Usage flows
+
+    /**
+     * <pre>
```

Comment:
you find html tags appealing ?  :D

Reply:
Yeap, because when you hover to the function it looks cool :D

Reply:
removed them